python/src/scipp/coords.py

```diff
@@ -0,0 +1,268 @@
+                self._del_coord(name)
+        if rename_dims:
+            blacklist = _get_splitting_nodes(self._rename)
+            for key, val in self._rename.items():
+                found = [k for k in key if k in self.obj.dims]
+                # rename if exactly one input is dimension-coord
+                if len(val) == 1 and len(
+                        found) == 1 and found[0] not in blacklist:
+                    self.obj = self.obj.rename_dims({found[0]: val[0]})
+        return self.obj
+
+
+def _get_splitting_nodes(graph):
+    nodes = {}
+    for key in graph:
+        for start in key:
+            nodes[start] = nodes.get(start, 0) + 1
+    return [node for node in nodes if nodes[node] > 1]
+
+
+def _transform_data_array(obj: DataArray, coords, graph: dict, *,
+                          kwargs) -> DataArray:
+    transform = CoordTransform(
+        obj,
+        graph=Graph(graph),
+        outputs=[coords] if isinstance(coords, str) else coords)
+    return transform.finalize(**kwargs)
+
+
+def _transform_dataset(obj: Dataset, coords, graph: dict, *,
+                       kwargs) -> Dataset:
+    # Note the inefficiency here in datasets with multiple items: Coord
+    # transform is repeated for every item rather than sharing what is
+    # possible. Implementing this would be tricky and likely error-prone,
+    # since different items may have different attributes. Unless we have
+    # clear performance requirements we therefore go with the safe and
+    # simple solution
+    return Dataset(
+        data={
+            name: _transform_data_array(
+                obj[name], coords=coords, graph=graph, kwargs=kwargs)
+            for name in obj
+        })
+
+
+def transform_coords(x: Union[DataArray, Dataset],
+                     coords: Union[str, List[str]],
+                     graph: CoordTransform.Graph,
+                     *,
+                     rename_dims=True,
+                     include_aliases=True,
+                     keep_intermediate=True,
+                     keep_inputs=True) -> Union[DataArray, Dataset]:
+    """Compute new coords based on transformation of input coords.
+
+    :param x: Input object with coords.
+    :param coords: Name or list of names of desired output coords.
+    :param graph: A graph defining how new coords can be computed from existing
+                  coords. This may be done in multiple steps.
+    :param rename_dims: Rename dimensions if products of dimension coord are
+                        fully consumed and consumer consumes exectly one
+                        dimension coordinate. Default is True.
+    :param include_aliases: If True, aliases for coords defined in graph are
+                            included in the output. Default is False.
+    :param keep_intermediate: Keep attributes created as intermediate results.
+                              Default is True.
+    :param keep_inputs: Keep consumed input coordinates or attributes.
```

Comment:
But we offer no finer control of which inputs or intermediates we might like to keep? I can't think of a use case for this at present.

Reply:
`del transformed.attr['unwanted']` should be enough

Reply:
Sounds ok to me. Unless there are some significant memory problems, I don't see this as being a problem.